earthaccess/results.py

```diff
@@ -107,6 +109,41 @@ def get_umm(self, umm_field: str) -> Union[str, Dict[str, Any]]:
         """
         return self["umm"].get(umm_field, "")
 
+    def doi(self) -> str | None:
+        """Retrieve the Digital Object Identifier (DOI) for this collection.
+
+        Returns:
+            This collection's DOI information, or `None`, if it has none.
+        """
+        doi = self["umm"].get("DOI", {})
+        if isinstance(doi, dict):
+            return doi.get("DOI", None)
+        return None
+
+    def citation(self, *, format: str, language: str) -> str | None:
+        """Generate a formatted citation for this collection using its DOI.
+
+        Parameters:
+            format: Citation format style (e.g., 'apa', 'bibtex', 'ris').
+                 For a full list of valid formats, visit <https://citation.doi.org/>
+            language: Language code (e.g., 'en-US').
+                 For a full list of valid language codes, visit <https://citation.doi.org/>
+
+        Returns:
+             The formatted citation as a string, or `None`, if this collection does not have a DOI.
+
+        Raises:
+             requests.RequestException: if fetching citation information from citations.doi.org failed.
+        """
+        if not self.doi():
+            return None
+
+        response = requests.get(
+            f"https://citation.doi.org/format?doi={self.doi()}&style={format}&lang={language}"
```

Comment:
Sorry, I missed this earlier, but prefer passing `params` rather than manually constructing the URL with query params:

```suggestion
            "https://citation.doi.org/format",
            params={"doi": self.doi(), "style": format, "lang": language},
```